### PR TITLE
Enable most Nuclei templates and remove obsolete checks

### PR DESCRIPTION
**Pull request by gguillemas**
This PR prepares the Nuclei check for more exhaustive scanning behavior by default and removes the checks that will be made obsolete by an existing Nuclei template in doing so. Rationale for each check below.

- **vulcan-tls** and **vulcan-certinfo** are replaced by various templates in the [SSL category](https://github.com/projectdiscovery/nuclei-templates/tree/main/ssl). The former check is also based on the long unmaintained [Mozilla Cipherscan](https://github.com/mozilla/cipherscan) tool with some questionable customization.
- **vulcan-exposed-files** and **vulcan-exposed-http-resources** are both very similar and are superceded in both flexibility and exhaustiveness of the database by the templates in the [exposed panels](https://github.com/projectdiscovery/nuclei-templates/tree/main/exposed-panels) and [exposures](https://github.com/projectdiscovery/nuclei-templates/tree/main/exposures) categories.
- **vulcan-s3-takeover** is replaced by the [AWS Bucket Takeover template](https://github.com/projectdiscovery/nuclei-templates/blob/main/takeovers/aws-bucket-takeover.yaml). The whole [takeovers](https://github.com/projectdiscovery/nuclei-templates/tree/main/takeovers) category also covers other takeover cases (e.g. Netlify, Wordpress.com...) which were not previously covered by Vulcan.
- **vulcan-lucky** checks for CVE-2016-2107 by some [weak ciphersuites](https://github.com/FiloSottile/CVE-2016-2107/blob/master/LuckyMinus20/CVE-2016-2107.go#L21-L26) which are detected by the [weak ciphersuintes](https://github.com/projectdiscovery/nuclei-templates/blob/main/ssl/weak-cipher-suites.yaml#L26-L221) template.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,27 +29,34 @@ Currently there's no vendoring provided for this project.
...
+* **[vulcan-exposed-files](https://github.com/adevinta/vulcan-checks/tree/3fb95a67d121a2993bd5db5eb4f7da0e57e31920/cmd/vulcan-exposed-files)** - Check asset for sensitive files exposed on HTTP server
+* **[vulcan-exposed-http-resources](https://github.com/adevinta/vulcan-checks/tree/3fb95a67d121a2993bd5db5eb4f7da0e57e31920/cmd/vulcan-exposed-http-resources)** - Checks if a web address exposes sensitive resources
+* **[vulcan-lucky](https://github.com/adevinta/vulcan-checks/tree/3fb95a67d121a2993bd5db5eb4f7da0e57e31920/cmd/vulcan-lucky)** - Checks if an TLS asset is vulnerable to LuckyMinus20 attack
+* **[vulcan-s3-takeover](https://github.com/adevinta/vulcan-checks/tree/3fb95a67d121a2993bd5db5eb4f7da0e57e31920/cmd/vulcan-s3-takeover)** - Checks for a vulnerability related to domain names pointing to a S3 static website when the S3 bucket has been deleted
+* **[vulcan-tls](https://github.com/adevinta/vulcan-checks/tree/3fb95a67d121a2993bd5db5eb4f7da0e57e31920/cmd/vulcan-tls)** - Analyzes TLS health of an asset
+
```
[the remark this answers is not in the document]
Comment (gguillemas) [reply README.md:59]: Yes, I realized while doing this. I will do it as a commit on the main PR after merging the others 👍

File: cmd/vulcan-nuclei/main.go [hunk 2]
```diff
@@ -281,79 +295,48 @@ func buildNucleiScanCmdArgs(target string, severities []string) []string {
...
-
-	return nucleiArgs
-}
-
-func generateTemplateSelectionList(skipUpdateTemplates bool, inclusionList, exclusionList []string) ([]string, error) {
-	if !skipUpdateTemplates {
-		_, err := runNucleiCmd([]string{"-update-templates"})
-		if err != nil {
-			return []string{}, errors.New("nuclei update-templates command execution failed")
-		}
```
[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-nuclei/main.go:301]: The other way around, now templates are never updates upon running the check, only at build time. However, it was not intentionally that I removed the option to configure this altogether, I will add this back with a default to "true". Thank you!

File: cmd/vulcan-nuclei/main.go [hunk 3]
```diff
@@ -281,79 +295,48 @@ func buildNucleiScanCmdArgs(target string, severities []string) []string {
...
+		tArg := []string{"-t", t}
+		nucleiArgs = append(nucleiArgs, tArg...)
 	}
 
-	logger.Debugf("templates path: %s%s", nucleiTemplatePath, nucleiTemplateListJSON)
-	tl, err := templateList()
-	if err != nil {
-		return []string{}, errors.New("unable to obtain nuclei template list")
-	}
-	availableTemplates := make(map[string]void)
```
[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-nuclei/main.go:309]: Not with the new mechanism, we previously did this manually and now we are delegating this to the Nuclei CLI.

File: cmd/vulcan-nuclei/main.go [hunk 4]
```diff
@@ -281,79 +295,48 @@ func buildNucleiScanCmdArgs(target string, severities []string) []string {
...
-	for _, t := range tl.Directory {
-		availableTemplates[t.Name] = void{}
+	// Exclude selected templates.
+	if len(opt.TemplateExclusionList) > 0 {
+		et := strings.Join(opt.TemplateExclusionList, ",")
+		logger.Infof("excluded templates: %s", et)
+		etArg := []string{"-et", et}
+		nucleiArgs = append(nucleiArgs, etArg...)
```
[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-nuclei/main.go:318]: Exactly. The CLI allows us to not only filter based on directories but also tags and allows us to remove quite a bit of code.

File: cmd/vulcan-nuclei/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (julianvilas) [reply cmd/vulcan-nuclei/main.go:309]: But what happens for example if I put a template like `fuzzzzzzz`, that doesn't exist? Is it ignored by the cli or it makes the check fail?

[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-nuclei/main.go:309]: Exactly. Here is a test I just did:
```
# nuclei -duc -target 172.17.0.1 -c 20 -json -vv -no-meta -H x-vulcan-nuclei -t network -et fuzzing,helpers,workflows -etags intrusive,fuzz,dos

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.6

                projectdiscovery.io

[INF] Using Nuclei Engine 2.8.6 (outdated)
[INF] Using Nuclei Templates 9.3.6 (latest)
[INF] Templates added in last update: 30
[INF] Templates loaded for scan: 70
[INF] Targets loaded for scan: 1
[...]
```

And now with nonsense arguments:
```
# nuclei -duc -target 172.17.0.1 -c 20 -json -vv -no-meta -H x-vulcan-nuclei -t network -et fuzzing,helpers,workflows,template-does-not-exist -etags intrusive,fuzz,dos,tag-does-not-exist

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.6

                projectdiscovery.io

[INF] Using Nuclei Engine 2.8.6 (outdated)
[INF] Using Nuclei Templates 9.3.6 (latest)
[INF] Templates added in last update: 30
[INF] Templates loaded for scan: 70
[INF] Targets loaded for scan: 1
```

As you can see the same number of templates run.

This also means that whenever we add a new exception we need to test that we have not mistyped and that the number of templates loaded for the scan has diminished after accounting for the new exclusion.

File: cmd/vulcan-nuclei/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-nuclei/main.go:301]: Changed in [434709e](https://github.com/adevinta/vulcan-checks/pull/247/commits/434709e7e3e0fbc0f89297e7c42af18930826ad1).

File: cmd/vulcan-nuclei/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (julianvilas) [reply cmd/vulcan-nuclei/main.go:309]: Gotcha :)

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (julianvilas) [reply README.md:59]: OK

File: cmd/vulcan-nuclei/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (julianvilas) [reply cmd/vulcan-nuclei/main.go:301]: Does the `-ut` prevail over the `-duc`?

[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-nuclei/main.go:301]: I just tested it and yes, it does prevail. However, in the test I found that if "-ut" is present Nuclei just checks for the updates but does not run the checks, I will push a fix and comment on this thread.

[the remark this answers is not in the document]
Comment (gguillemas) [reply cmd/vulcan-nuclei/main.go:301]: Fixed in [e2f27c3](https://github.com/adevinta/vulcan-checks/pull/247/commits/e2f27c321a74c93e2b1695fdfd57e6e790ade229).

[the remark this answers is not in the document]
Comment (julianvilas) [reply cmd/vulcan-nuclei/main.go:301]: Cool!

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (julianvilas) [reply README.md:59]: Should I hold approval until this is actually done?

[the remark this answers is not in the document]
Comment (gguillemas) [reply README.md:59]: You can approve whenever you are ready and I will commit after merging this and the other PRs and before merging the main PR on master.

**Review by julianvilas — APPROVED** [review]

**Pull request merged**